### DISCUSSION TITLE
Possibility to specify Authentication requirements per listener

### DISCUSSION THREAD

**Discussion by realdognose** [nanomq/nanomq#1536] [Ideas]
I'm currently playing around with bridging to solve an Issue. 

However, derived from that, what would actually solve my issue in an way easier way would be to be able to configure authentication
settings based on listener. 

Background: I have a site local mqtt-server, which is allowing non-https and non-authenticated connections. For some clients I now need to be able to connect through the internet, but want to avoid to have all the authentication and encryption overhead for local usage. 

So, what could easily solve this would be to be able to specify authentication settings per listener, like this: 

```
listeners.tcp {
    bind = "0.0.0.0:1883"
	
	auth {
		allow_anonymous = true
		no_match = allow
		deny_action = ignore

		cache = {
			max_size = 32
			ttl = 1m
		}
	}
}

listeners.ssl {
    bind = "0.0.0.0:8883"
    keyfile = "/etc/certs/key.pem"
    certfile = "/etc/certs/cert.pem"
    cacertfile = "/etc/certs/cacert.pem"
    verify_peer = false
    fail_if_no_peer_cert = false
	
	auth {
		allow_anonymous = false
		no_match = deny
		deny_action = ignore

		cache = {
			max_size = 32
			ttl = 1m
		}

		password = {include "/etc/nanomq_pwd.conf"}
		acl = {include "/etc/nanomq_acl.conf"}
	}
}
```
the global configuration could remain and apply, unless explicitly overwritten by listener-specific settings.

I guess, code wise this would just be something like 

```
if (exists(configuration[thisListener].auth)){
    configureAuth(thisListener, configuration[thisListener].auth)
}else{
    configureAuth(thisListener, configuration.auth)
}
```

**Comment by JaylinYu** [comment]
Totally understood, duplicate with https://github.com/nanomq/nanomq/discussions/1342
just not our first priority to implement this.....
happy to accept any PR about this....

**Reply by realdognose** [reply]
not exactly a duplicate, but same basic requirement (more flexibility for individual listener configuration)

**Reply by JaylinYu** [reply]
Sure, will make this happen
